tracer-core/src/main/java/org/zalando/tracer/FlowIDGenerator.java

```diff
@@ -2,28 +2,27 @@
 
 import org.apiguardian.api.API;
 
-import java.nio.ByteBuffer;
 import java.util.Base64;
-import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.apiguardian.api.API.Status.STABLE;
 
 @API(status = STABLE)
 public final class FlowIDGenerator implements Generator {
 
-    private final Base64.Encoder encoder = Base64.getUrlEncoder();
+    private static final int LENGTH = Long.BYTES * 2;
+    private static final byte R = "R".getBytes(UTF_8)[0];
+
+    private final Base64.Encoder encoder = Base64.getUrlEncoder().withoutPadding();
 
     @Override
     public String generate() {
-        final UUID uuid = UUID.randomUUID();
-
-        final ByteBuffer buffer = ByteBuffer.wrap(new byte[16]);
-        buffer.putLong(uuid.getMostSignificantBits());
-        buffer.putLong(uuid.getLeastSignificantBits());
-
-        final byte[] bytes = buffer.array();
-
-        return "R" + encoder.encodeToString(bytes).replaceAll("=", "").substring(1);
+        final byte[] bytes = new byte[LENGTH];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        final byte[] encoded = encoder.encode(bytes);
+        encoded[0] = R;
+        return new String(encoded, UTF_8);
```

Comment:
Makes no sense to use utf-8 since it is guaranteed to be single byte encoding - it should be StandardCharsets.ISO_8859_1

Reply:
Any superset of ASCII will work. `encoded` is gaarantued to contain ASCII only.

>  A 65-character subset of US-ASCII is used
>
> https://tools.ietf.org/html/rfc4648#section-4

Reply:
Sure, but charset encoding have to inspect bytes to see whether it is multi-byte while ISO probably just uses mapping table. Would be  interesting to micro-benchmark this :)